### PR TITLE
Fix `typing.TypeAliasType` being undefined on python < 3.12

### LINKED ISSUE #17554: mypy 1.11.0 `Could not find builtin symbol 'TypeAliasType'`
```
/Users/claudiumuresan/opt/anaconda3/envs/waylay_query/lib/python3.11/site-packages/waylay/sdk/api/_models.py:196: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.11.0
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 5830, in accept
  File "mypy/nodes.py", line 2714, in accept
  File "mypy/checkexpr.py", line 4730, in visit_type_alias_expr
  File "mypy/checkexpr.py", line 4749, in alias_type_in_runtime_context
  File "mypy/checkexpr.py", line 5863, in named_type
  File "mypy/checker.py", line 6924, in named_type
  File "mypy/checker.py", line 7041, in lookup_qualified
KeyError: "Could not find builtin symbol 'TypeAliasType' (If you are running a test case, use a fixture that defines this symbol)"
/Users/claudiumuresan/opt/anaconda3/envs/waylay_query/lib/python3.11/site-packages/waylay/sdk/api/_models.py:196: : note: use --pdb to drop into pdb
make: *** [typecheck] Error 1
```

### PR DESCRIPTION
**Pull request by sobolevn**
Closes #17554
CC @JukkaL 
Refs https://github.com/python/mypy/pull/17320

### CODE REVIEW

File: test-data/unit/check-type-aliases.test [hunk 1]
```diff
@@ -1084,6 +1084,15 @@ yc: A.ClassAlias = ""  # E: Incompatible types in assignment (expression has typ
 [builtins fixtures/tuple.pyi]
 [typing fixtures/typing-full.pyi]
 
+[case testTypeAliasTypePython311]
+# flags: --python-version 3.11
+# Pinning to 3.11, because 3.12 have `TypeAliasType`
```
Comment (JelleZijlstra) [inline test-data/unit/check-type-aliases.test:1089]:
```suggestion
# Pinning to 3.11, because 3.12 has `typing.TypeAliasType`
```

**Review by JelleZijlstra — APPROVED** [review]

File: test-data/unit/check-type-aliases.test (hunk 1 above)
Comment (sobolevn) [reply test-data/unit/check-type-aliases.test]: Thanks!

**Review by hauntsaninja — APPROVED** [review]

**Comment by toondaey** [comment]
This doesn't seem to fix the issue with `Could not find builtin symbol TypeAliasType` or has the release for this been delayed?

**Comment by sobolevn** [comment]
@toondaey please, try `main` branch version. `1.11.1` was not yet released.

**Pull request merged**